dGame/UserManager.cpp

```diff
@@ -268,24 +267,40 @@ void UserManager::CreateCharacter(const SystemAddress& sysAddr, Packet* packet)
 	User* u = GetUser(sysAddr);
 	if (!u) return;
 
-	std::string name = PacketUtils::ReadString(8, packet, true);
-
-	uint32_t firstNameIndex = PacketUtils::ReadU32(74, packet);
-	uint32_t middleNameIndex = PacketUtils::ReadU32(78, packet);
-	uint32_t lastNameIndex = PacketUtils::ReadU32(82, packet);
+	CINSTREAM_SKIP_HEADER;
+	LUWString LUWStringName(33);
+	inStream.Read(LUWStringName);
+	const auto name = LUWStringName.GetAsString();
+
+	uint32_t firstNameIndex;
+	inStream.Read(firstNameIndex);
+	uint32_t middleNameIndex;
+	inStream.Read(middleNameIndex);
+	uint32_t lastNameIndex;
+	inStream.Read(lastNameIndex);
 	std::string predefinedName = GetPredefinedName(firstNameIndex, middleNameIndex, lastNameIndex);
 
-	uint32_t shirtColor = PacketUtils::ReadU32(95, packet);
-	uint32_t shirtStyle = PacketUtils::ReadU32(99, packet);
-	uint32_t pantsColor = PacketUtils::ReadU32(103, packet);
-	uint32_t hairStyle = PacketUtils::ReadU32(107, packet);
-	uint32_t hairColor = PacketUtils::ReadU32(111, packet);
-	uint32_t lh = PacketUtils::ReadU32(115, packet);
-	uint32_t rh = PacketUtils::ReadU32(119, packet);
-	uint32_t eyebrows = PacketUtils::ReadU32(123, packet);
-	uint32_t eyes = PacketUtils::ReadU32(127, packet);
-	uint32_t mouth = PacketUtils::ReadU32(131, packet);
-
+	inStream.IgnoreBytes(9);
+	uint32_t shirtColor;
```

Comment:
Since these are all uint32_t, a single declaration with all the names and the a plain sequence of Read calls may be easier to read.

Reply:
true